satpy/modifiers/angles.py

```diff
@@ -250,6 +248,7 @@ def _hash_args(*args, unhashable_types=DEFAULT_UNCACHE_TYPES):
     hashable_args = []
     for arg in args:
         if isinstance(arg, unhashable_types):
+            warnings.warn(f"Unhashable type in function signature ({type(arg)}), cannot be cached.", stacklevel=2)
```

Comment:
I'm unsure about this. For one the stacklevel should be at least 3 in my opinion as 2 only gets us to `__call__` of this helper class.

As for warning versus something else, I think it is confusing. It is perfectly valid to call these functions with an uncacheable/unhashable type like a SwathDefinition or a StackedAreaDefinition. There shouldn't be a warning for these cases. In my previous comment I pointed out that this warning needs to go *after* the `should_cache` check which checks if the user has caching enabled. Otherwise users could be getting this when they know they aren't going to cache things (uncacheable types) and they haven't even requested that it be attempted (the config key is disabled/off).

Reply:
Good point! Let me fix that.

Reply:
This is now fixed. Thanks for pointing this out.